documentation/PSPChannel.md

```diff
@@ -0,0 +1,334 @@
+
+the PSPChannle client can request various actions by
+transmitting the specified data to the server.
+
+```js
+socket.send(data);
+```
+
+* Parameters:
+
+ **data**: {Object}. The data to send to the server. It contains two paths: `type` and `value`.
+
+ **data.type**: {String}. It can be one of these values: `modelChanged`, `pullModel`.
+
+ _`modelChanged`_ typed messages are used to apply preferences, switch to a different preference set and save changed preferences.
+
+ _`pullModel`_ typed messages are used to read preference values.
+
+ **data.value**: {Object}. The actual value for performing actions. The detailed value structure is described below:
+
+#### Apply a Preference
+
+* The data transmitted from the client to the server:
+
+An example of changing the volume:
+
+```json
+{
+    "type": "modelChanged",
+    "value": {
+        "settingControls": {
+            "http://registry\\.gpii\\.net/common/volume": {
+                "value": 0.5
+             }
+         }
+    }
+}
+```
+
+ An example of changing a nested keyed preference:
+
+```json
+{
+    "type": "modelChanged",
+    "value": {
+        "settingControls": {
+            "http://registry\\.gpii\\.net/applications/com\\.microsoft\\.office.word-ribbon": {
+                "value": "StandardSet"
+             }
+         }
+    }
+}
+```
+
+* The server response:
+
+When preferences are applied, the server will send back a `modelChanged` message. An example:
+
+```json
+{
+    "type": "modelChanged",
+    "payload": {
+        "path": [],
+        "type": "ADD",
+        "value": {
+            "gpiiKey": "noUser",
+            "activeContextName": "gpii-default",
+            "settingControls": {
+                "http://registry\\.gpii\\.net/common/volume": {
+                    "value": 0.5,
+                    "schema": {
+                        "title": "Volume",
+                        "description": "General volume of the operating system",
+                        "type": "number",
+                        "minimum": 0,
+                        "maximum": 1
+                    },
+                    "liveness": "live"
+                }
+            },
+            "preferences": {
+                "contexts": {
+                    "gpii-default": {}
+                }
+            }
+        }
+    }
+}
+```
+
+#### Read a Preference
+
+* The data transmitted from the client to the server:
+1. An example of reading the magnification value:
+
+```json
+{
+    "type": "pullModel",
+    "value": {
+        "settingControls": {
+            "http://registry\\.gpii\\.net/common/volume": {
+                "value": 1
+            }
+        }
+    }
+}
+```
+
+The value at the path `value.settingControls.{preference}.value` can be any valid value constrained by this setting's
+schema defined in its solution registry entry. This value is required for sending the requested preference through the
+matchmaking process but it doesn't take any effect to the system.
```

Comment:
take any effect -> "have any effect"

Reply:
"to the system" -> "on the system"
After the changes, the phrase is "have any effect on the system",